src/parser/uniterm.rs

```diff
@@ -207,38 +244,26 @@ impl UniRecord {
                     } else {
                         let elem = path.pop().unwrap();
                         match elem {
-                            FieldPathElem::Ident(id) => {
-                                // At parsing stage, all `Rc`s must be 1-counted. We can thus call
-                                // `into_owned()` without risking to actually clone anything.
-                                match rt.term.into_owned() {
-                                    Term::MetaValue(MetaValue {
-                                        doc: None,
-                                        types: Some(ctrt),
-                                        contracts,
-                                        opt: false,
-                                        priority: MergePriority::Neutral,
-                                        value: None,
-                                    }) if contracts.is_empty() => {
-                                        Ok(RecordRows(RecordRowsF::Extend {
-                                            row: RecordRow {
-                                                id,
-                                                types: Box::new(ctrt.types),
-                                            },
-                                            tail: Box::new(acc),
-                                        }))
-                                    }
-                                    _ => {
-                                        // Position of identifiers must always be set at this stage
-                                        // (parsing)
-                                        let span_id = id.pos.unwrap();
-                                        let term_pos = rt.pos.into_opt().unwrap_or(span_id);
-                                        Err(InvalidRecordTypeError(TermPos::Original(
-                                            RawSpan::fuse(span_id, term_pos).unwrap(),
-                                        )))
-                                    }
-                                }
+                            FieldPathElem::Ident(id) => term_to_record_rows(id, rt, acc),
+                            FieldPathElem::Expr(expr) => {
+                                let Term::StrChunks(chunks) = expr.term.as_ref() else {
+                                        return Err(InvalidRecordTypeError(rt.pos))
+                                    };
+                                let Some(id) =
+                                    chunks.iter().fold(Some(String::new()), |acc, next| {
```

Comment:
Looks like it has been copied verbatim from https://github.com/tweag/nickel/blob/5449b4c63f247cc804221f0f1cbdc865e8496784/src/parser/utils.rs#L246. I think it's worth making it a function, such as `StrChunks::try_as_static_str(&self) -> Option<String>` and replace the use sites by it.